pkg/infrastructure/azure/azure.go

```diff
@@ -738,14 +742,33 @@ func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]
 		blobIgnitionContainer = createBlobContainerOutput.BlobContainer
 		logrus.Debugf("BlobIgnitionContainer.ID=%s", *blobIgnitionContainer.ID)
 	}
-
 	sasURL := ""
+	now := time.Now().UTC().Add(-10 * time.Second)
+	expiry := now.Add(1 * time.Hour)
+	info := service.KeyInfo{
+		Start:  to.Ptr(now.UTC().Format(sas.TimeFormat)),
+		Expiry: to.Ptr(expiry.UTC().Format(sas.TimeFormat)),
+	}
+
+	serviceClient, err := service.NewClient(fmt.Sprintf("https://%s.blob.core.windows.net/", p.StorageAccountName),
+		session.TokenCreds,
+		&service.ClientOptions{},
```

Comment:
We need to set this to switch between govcloud and public cloud
```suggestion
		&service.ClientOptions{
			ClientOptions: azcore.ClientOptions{
				Cloud: p.CloudConfiguration,
			},
		},
```